### PR TITLE
07.POST/api/articles/article_id/comments

### PR DESCRIPTION
**Pull request by thebonottox**

### CODE REVIEW

**Review by MrMcSnail — APPROVED** [review]
Merge away

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -241,3 +241,61 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+      .post("/api/articles/9/comments")
+      .expect(201)
+      .send({
+        username: "butter_bridge",
+        body: "I have a body",
+      })
+      .then((response) => {
+        const body = response.body;
+        expect(body).toEqual({
+          username: "butter_bridge",
```
Comment (two-triangles) [inline __tests__/endpoint.test.js:259]: comments in the database should contain the following properties: 

 - comment_id
 - votes
 - created_at
 - author which is the username from the users table
- body

File: __tests__/endpoint.test.js [hunk 2]
```diff
@@ -241,3 +241,61 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+      .post("/api/articles/9999/comments")
+      .expect(404)
+      .send({
+        username: "butter_bridge",
+        body: "I have a body",
+      })
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article ID not found.");
+      });
+  });
```
Comment (two-triangles) [inline __tests__/endpoint.test.js:300]: some tests still needed : 

400 - missing properties on sent in object
201 -  posts a comments and ignores any extra properties

File: app/controllers/controllers.js [hunk 3]
```diff
@@ -43,14 +44,31 @@ const getCommentsByArticleId = (request, response, next) => {
...
 };
-//---------------------------------------------
 
-module.exports = { getTopics, getArticles,getArticleById,getCommentsByArticleId };
+const postComment = (request, response, next) => {
+  const { article_id } = request.params;
+  const newComment = request.body;
+  addComment(newComment, article_id)
+    .then((rows) => {
+      const sendBack = { username: rows[0].author, body: rows[0].body };
```
Comment (two-triangles) [inline app/controllers/controllers.js:59]: see comment in tests about what should be sent back from the server

**Pull request merged**